internal/log/log.go

```diff
@@ -36,12 +36,15 @@ var (
 	logger ddtrace.Logger = &defaultLogger{l: log.New(os.Stderr, "", log.LstdFlags)}
 )
 
-// UseLogger sets l as the active logger.
-func UseLogger(l ddtrace.Logger) {
+// UseLogger sets l as the active logger and returns the previously configured
+// logger.
+func UseLogger(l ddtrace.Logger) ddtrace.Logger {
```

Comment:
Since this only makes sense in tests (returning the old logger), why not:
```suggestion
// UseLogger sets l as the active logger and returns a function to restore the previous
// logger. The return value is mostly useful when testing.
func UseLogger(l ddtrace.Logger) (restore func()) {
```
Then, in tests, you could do:
```go
defer UseLogger(rc)()
```
Much shorter and cleaner imho.

Reply:
Done in bce3c86482fd0103e6109458412ea4ff59e6d487. I considered this but figured it might be overfitting the API to a singular use case. That being said, since we only have this use case for now and it's easy to iterate on an internal package like this, why not : )?